tests/test_baggage.py

```diff
@@ -0,0 +1,161 @@
+        if isinstance(headers, dict):
+            assert "baggage" not in headers
+        # When headers is a list, ensure no header entry has key "baggage"
+        elif isinstance(headers, list):
+            assert all(header.get("key") != "baggage" for header in headers)
+
+
+@scenarios.default
+@features.datadog_baggage_headers
+class Test_Baggage_Headers_Malformed2:
+    def setup_main(self):
+        self.r = weblog.get(
+            "/make_distant_call",
+            params={"url": "http://weblog:7777"},
+            headers={"x-datadog-parent-id": "10", "x-datadog-trace-id": "2", "baggage": "=no-key"},
+        )
+
+    def test_main(self):
+        interfaces.library.assert_trace_exists(self.r)
+        assert self.r.status_code == 200
+        data = json.loads(self.r.text)
+        headers = data["request_headers"]
+        if isinstance(headers, dict):
+            assert "baggage" not in headers
+        elif isinstance(headers, list):
+            assert all(header.get("key") != "baggage" for header in headers)
+
+
+@scenarios.only_baggage_propagation
+@features.datadog_baggage_headers
+class Test_Only_Baggage_Header:
+    def setup_main(self):
+        self.r = weblog.get("/make_distant_call", params={"url": "http://weblog:7777"}, headers={"baggage": "foo=bar"})
+
+    def test_main(self):
+        interfaces.library.assert_trace_exists(self.r)
+        assert self.r.status_code == 200
+        data = json.loads(self.r.text)
+        baggage_value = extract_baggage_value(data["request_headers"])
+        assert baggage_value is not None
+        assert "foo=bar" in baggage_value
+
+
+@scenarios.default
+@features.datadog_baggage_headers
+class Test_Baggage_Headers_Max_Items:
+    def setup_main(self):
+        self.max_items = 64
+        baggage_items = [f"key{i}=value{i}" for i in range(self.max_items + 2)]
+        baggage_header = ",".join(baggage_items)
+        self.r = weblog.get(
+            "/make_distant_call",
+            params={"url": "http://weblog:7777"},
+            headers={
+                "x-datadog-parent-id": "10",
+                "x-datadog-trace-id": "2",
+                "x-datadog-sampling-priority": "1",
+                "baggage": baggage_header,
+            },
+        )
+
+    def test_main(self):
+        interfaces.library.assert_trace_exists(self.r)
+        assert self.r.status_code == 200
+        data = json.loads(self.r.text)
+        baggage_header_value = extract_baggage_value(data["request_headers"])
+        assert baggage_header_value is not None
+        header_str = baggage_header_value[0] if isinstance(baggage_header_value, list) else baggage_header_value
+        items = header_str.split(",")
+        # Ensure we respect the max items limit
+        assert len(items) <= self.max_items
+
+
+@scenarios.default
+@features.datadog_baggage_headers
+class Test_Baggage_Headers_Max_Bytes:
+    def setup_main(self):
+        self.max_bytes = 8192
+        baggage_items = {
```

Comment:
nit: I think this test case can be revised so it's even simpler and easier for authors. If you only test two keys and set the value lengths to `self.max_bytes // 2` it makes it much easier to understand that:
- 8192/2 will be even (no weird modulo's)
- It's more apparent that the length of the keys, equal signs, and delimiter will make the total exceed the max_bytes